src/firebase/fireStore.js

```diff
@@ -61,25 +61,7 @@ class FireStore {
         querySnapshotResults.push(promise);
       });
       // Get Location telemetry from Google Maps Api.
-      const hospitalRecordsResults = await Promise.all(querySnapshotResults);
-      const promiseData = hospitalRecordsResults.map(
-        async (hospitalRecords) => {
-          if (!!hospitalRecords) {
-            const mappedData = hospitalRecords.map(async (item) => {
-              const name = item.facility_type;
-              const response = await fetch(
-                `https://maps.googleapis.com/maps/api/geocode/json?address=${name},${cityName}&sensor=true&key=${GOOGLE_MAPS_API_KEY}`
-              );
-              results.push([item, response.json()]);
-            });
-            await Promise.all(mappedData);
-            return results;
-          }
-        }
-      );
-      // Merge each hospital records data with its geolocation meta and invoke callback.
-      await Promise.all(promiseData);
-      const mappedData = results.map((data) => {
+      const mappedData = querySnapshotResults.map((data) => {
         /** @type {Array<Object<String, String>, Promise>} */
         const [item, promise] = data;
```

Comment:
@CarlKarlQarl i think we can refactor this better now, we can remove L64-74 too;

```javascript
        /** @type {Array<Object<String, String>, Promise>} */
        const [item, promise] = data;
```
Remove this.

Instead where we do `querySnapshot.forEach`, on L58-60 and on [L61](https://github.com/Xcov19/mycovidconnect/blob/752a2ecdde86026ec71024eb95e3bb209daa7f41/src/firebase/fireStore.js#L61)  do this:

```javascript
const promise = new Promise((resolve) =>
         const records = JSON.parse(docData.records);
         if (records.length) {
            // Called for each hospital result.
           results.push(records);
           resolve();
          }
        );
promisesArr.push(promise);

```

and finally:
```javascript
await Promise.all(promisesArr);
mapCallback(results);
```

Reply:
@vasynizami have a look at this pr; will discuss this soon.